docs/design/architecture.md

```diff
@@ -2,128 +2,353 @@
-
-For any given container, both the init process and all potentially executed
-commands within that container, together with their related I/O streams, need
-to go through the VSOCK interface exported by QEMU.
-
-The container workload, that is, the actual OCI bundle rootfs, is exported from the
-host to the virtual machine.  In the case where a block-based graph driver is
-configured, `virtio-scsi` will be used. In all other cases a `virtio-fs` VIRTIO mount point
-will be used. `kata-agent` uses this mount point as the root filesystem for the
-container processes.
+## System overview
 
-## Virtualization
+### Container creation
+
+The steps below show at a high level how a Kata Container container is
+created using the containerd container manager:
+
+1. The user requests the creation of a container by running `ctr run`.
+1. The container manager daemon runs a single instance of the Kata [runtime](#runtime).
+1. The Kata runtime loads its [configuration file](#configuration).
+1. The container manager calls a set of shimv2 API functions on the runtime.
+1. The Kata runtime launches the configured [hypervisor](#hypervisor).
+1. The hypervisor creates a VM using the [guest assets](#guest-assets).
+1. The hypervisor boots (starts) the VM.
+1. The [agent](#agent) is launched inside the VM.
+1. The agent creates a container environment to host the [workload](#workload).
+1. The agent spawns the workload inside the container environment.
+1. Control is returned to the user running `ctr`.
+
+   If the user requested it, the standard I/O streams of the workload
+   running inside the container in the VM are connected to the user's
+   terminal.
+
+Further details of these steps are provided in the sections below.
+
+### Container shutdown
 
-How Kata Containers maps container concepts to virtual machine technologies, and how this is realized in the multiple
-hypervisors and VMMs that Kata supports is described within the [virtualization documentation](./virtualization.md)
+There are two possible ways for the container environment to be
+destroyed:
+
+- When the [workload](#workload) exits.
+- When the container manager forces the container to be deleted.
+
+#### Workload exit
+
+The [agent](#agent) will detect when the [workload](#workload) process
+exits, capture its exit code and return it to the runtime, which
+passes it back to the container manager. Once the workload has fully
+exited, the VM is no longer needed and the runtime cleans up the
+environment (which includes killing the [hypervisor](#hypervisor)
+process).
+
+> **Note:**
+>
+> When [agent tracing is enabled](../tracing.md#agent-shutdown-behaviour),
+> the shutdown behaviour is different.
+
+#### Container manager requested shutdown
+
+If the container manager requests the container be deleted, the
+[runtime](#runtime) will signal the agent by sending it a
+[`DestroySandbox` ttRPC API](../../src/agent/protocols/protos/agent.proto) request.
```

Comment:
Since the link goes to the agent.proto file and not directly to the DestroySandbox line in it I would wrap the URL around "ttRPC API" only and not have the DestroySandbox text linkable.